solr/core/src/java/org/apache/solr/servlet/SolrDispatchFilter.java

```diff
@@ -209,8 +208,8 @@ public void doFilter(
     Tracer t = getCores() == null ? GlobalTracer.get() : getCores().getTracer();
     request.setAttribute(Tracer.class.getName(), t);
     RateLimitManager rateLimitManager = coreService.getService().getRateLimitManager();
-    request.setAttribute(RateLimitManager.class.getName(), rateLimitManager);
```

Comment:
It's not clear to me if the RateLimitManager was on the request merely out of convenience for the existing logic or if it was intentionally used for miscellaneous unforeseen use-cases (and you are now removing it).  Shrug.

Reply:
luckily we have very good tooling support to show where a class is used inside a project. I can 100% tell that this is not used anywhere else. Happy to be shown different. I think this was just a partial/unfinished plan to do something and it ended up in an awkward place where it is set as an attribute instead of simply passing the ref.
I took the liberty to remove some of the tech debt by cleaning up some code I saw was not clear, if this is putting too much burden on the review I will be happy to revert, let me know.

Reply:
Of course we can use Java tooling to look within Solr to see but this may be here for external / plugins.  In general we put objects on the request for plugins.  I'm not familiar with the RateLimitManager to consider if there would be a point for any plugin to want to get it from the request.  My intuition agrees with you.

Reply:
hmm. good point. I was quick to say it's not used in Solr, but you are of course right plugins are something I forgot about. still, the rate limiter doesn't strike me as being a good fit for a request attribute. I am ok with leaving this in if you think there is a risk of regression.